src/CFamily/CMake/CompilationDatabaseLocator.cs

```diff
@@ -52,68 +55,68 @@ internal class CompilationDatabaseLocator : ICompilationDatabaseLocator
-            IBuildConfigProvider activeConfigProvider, ILogger logger)
+        public CompilationDatabaseLocator(IFolderWorkspaceService folderWorkspaceService, 
+            IBuildConfigProvider buildConfigProvider,
+            IFileSystem fileSystem, 
+            ILogger logger)
         {
             this.folderWorkspaceService = folderWorkspaceService;
             this.fileSystem = fileSystem;
-            this.buildConfigProvider = activeConfigProvider;
+            this.buildConfigProvider = buildConfigProvider;
             this.logger = logger;
         }
 
         public string Locate()
         {
-            var databaseFolderPath = GetCompilationDatabaseFolderPath();
+            var rootDirectory = folderWorkspaceService.FindRootDirectory();
 
-            if (string.IsNullOrEmpty(databaseFolderPath))
+            if (string.IsNullOrEmpty(rootDirectory))
             {
+                logger.LogDebug(Resources.NoRootDirectory);
                 return null;
             }
 
-            var compilationDatabaseFilePath = Path.Combine(databaseFolderPath, CompilationDatabaseFileName);
+            var cmakeSettingsFullPath = Path.GetFullPath(Path.Combine(rootDirectory, CMakeSettingsFileName));
+            var activeConfiguration = buildConfigProvider.GetActiveConfig(rootDirectory);
 
-            if (!fileSystem.File.Exists(compilationDatabaseFilePath))
-            {
-                logger.WriteLine(Resources.NoCompilationDatabaseFile, compilationDatabaseFilePath);
+            return fileSystem.File.Exists(cmakeSettingsFullPath)
+                ? GetConfiguredLocation(cmakeSettingsFullPath, activeConfiguration, rootDirectory)
+                : GetDefaultLocation(rootDirectory, activeConfiguration);
+        }
 
-                return null;
-            }
+        private string GetDefaultLocation(string rootDirectory, string activeConfiguration)
+        {
+            var defaultDirectory = Path.GetFullPath(string.Format(DefaultLocationFormat, rootDirectory, activeConfiguration));
+            var defaultLocation = Path.Combine(defaultDirectory, CompilationDatabaseFileName);
 
-            logger.WriteLine(Resources.FoundCompilationDatabaseFile, compilationDatabaseFilePath);
+            logger.LogDebug(Resources.NoCMakeSettings, CMakeSettingsFileName, rootDirectory, defaultLocation);
 
-            return compilationDatabaseFilePath;
+            return defaultLocation;
         }
 
-        public string GetCompilationDatabaseFolderPath()
+        private string GetConfiguredLocation(string cmakeSettingsFullPath, string activeConfiguration, string rootDirectory)
         {
-            var rootDirectory = folderWorkspaceService.FindRootDirectory();
+            logger.LogDebug(Resources.ReadingCMakeSettings, cmakeSettingsFullPath);
+            CMakeSettings settings;
 
-            if (string.IsNullOrEmpty(rootDirectory))
+            try
```

Comment:
The try-catch was missing before. I decided to catch it here, instead of in the caller, and added new tests.